### DISCUSSION TITLE
Wrapper Object to use piper directly within process

### DISCUSSION THREAD

**Discussion by cjohn001** [rhasspy/piper#411] [Ideas]
Hello together, I am currently playing around with piper, as I am searching for a good tts solutions for rapid prototyping in Python which can be easily ported to mobile applications for Android and iOS. So far it seems to me that piper needs to be used via command line in a dedicated process or as an external server process. I also found the explanation here https://hackernoon.com/how-to-set-up-piper-speech-synthesizer-on-a-local-machine to attach to it via pipes. However, I am wondering if their is also a direct usage option for the library available currently. I am suspecting that their might already be some kind of cython bindings for usage with python and a more direct approach to use piper in custom applications. In case such an interface is available, it would be great if you could provide directions. Otherwise, I am wondering if such a feature like is already on the roadmap. Thanks for your feedback. It is a great product you are currently building!

**Comment by csukuangfj** [comment]
>  I am searching for a good tts solutions for rapid prototyping in Python

> which can be easily ported to mobile applications for Android and iOS.


I suggest that you have a look at
https://github.com/k2-fsa/sherpa-onnx

It supports all models trained by piper. At the same time, it provides Python APIs and supports Android and iOS.

You can find pre-built Android APKs with piper models at
https://k2-fsa.github.io/sherpa/onnx/tts/apk-engine.html

**Comment by cjohn001** [comment]
@csukuangfj Thanks for the directions. I will have a look at it.
